### PR TITLE
feat(install): add recipe validation

### PR DESCRIPTION
**Pull request by ctrombley**

### CODE REVIEW

File: internal/install/recipe_file.go [hunk 1]
```diff
@@ -6,11 +6,11 @@ type recipeFile struct {
 	Install        map[string]interface{} `yaml:"install"`
 	InstallTargets recipeInstallTarget    `yaml:"installTargets"`
 	Keywords       []string               `yaml:"keywords"`
-	MELTMatch      meltMatch              `yaml:"meltMatch"`
```
Comment (zlesnr) [inline internal/install/recipe_file.go:9]: Last conversation we had, I thought this was remaining.  Am I out of date here?

**Review by zlesnr — APPROVED** [review]

File: internal/install/go_task_recipe_executor.go [hunk 2]
```diff
@@ -117,14 +86,32 @@ func executeRecipeSteps(ctx context.Context, m discoveryManifest, r recipeFile)
 	return nil
 }
 
+func setProfileVars(t *taskfile.Taskfile) error {
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply internal/install/go_task_recipe_executor.go:89]: 👍

File: internal/install/mock_nrdb_client.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	}, nil
+}
+
+func (c *mockNrdbClient) ThrowError(message string) {
+	c.error = message
+}
+
+func (c *mockNrdbClient) ReturnResultsAfterNAttempts(results []nrdb.NrdbResult, attempts int) {
+	c.results = func() []nrdb.NrdbResult {
+		fmt.Printf("c.attempts: %d\n", c.attempts)
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply internal/install/mock_nrdb_client.go:43]: This was a mistake, I'll remove it.

File: internal/install/polling_recipe_validator.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+func (m *pollingRecipeValidator) validate(ctx context.Context, r recipe) (bool, error) {
+	count := 0
+	ticker := time.NewTicker(m.interval)
+	defer ticker.Stop()
+
+	resultChan := make(chan bool)
+	errChan := make(chan error)
+
+	go func() {
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply internal/install/polling_recipe_validator.go:42]: Totally right, good call!  I'll clean it up.

File: internal/install/polling_recipe_validator.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+		for {
+			if count == m.maxAttempts {
+				resultChan <- false
+				return
+			}
+
+			log.Debugf("Validation attempt #%d...", count+1)
+			ok, err := m.tryValidate(ctx, r)
+			if err != nil {
+				errChan <- err
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply internal/install/polling_recipe_validator.go:52]: This is getting cleaned up.

File: internal/install/recipe_file.go (hunk 1 above)
Comment (ctrombley) [reply internal/install/recipe_file.go:9]: You are right, @jbeveland27 says it's staying for now.  For some reason it was causing panics when marshaling to YAML the way it was set up before.  I'll put it back in place.

File: internal/install/polling_recipe_validator.go [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	v := pollingRecipeValidator{
+		maxAttempts: defaultMaxAttempts,
+		interval:    defaultInterval,
+		client:      c,
+	}
+
+	return &v
+}
+
+func (m *pollingRecipeValidator) validate(ctx context.Context, r recipe) (bool, error) {
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply internal/install/polling_recipe_validator.go:34]: Leaving as-is per our conversation.

**Review by zlesnr — APPROVED** [review]

**Pull request merged**